packages/deployer/subtasks/deploy-proxy.js

```diff
@@ -5,29 +5,22 @@ const { SUBTASK_DEPLOY_PROXY, SUBTASK_DEPLOY_CONTRACT } = require('../task-names
 subtask(SUBTASK_DEPLOY_PROXY, 'Deploys the main proxy if needed').setAction(async (_, hre) => {
   logger.subtitle('Deploying main proxy');
 
-  const routerName = 'Router';
-  const proxyName = hre.config.deployer.proxyContract;
-
-  await _deployProxy(proxyName, routerName, hre);
-
-  _setProxyOnModules(proxyName, hre);
-});
-
-async function _deployProxy(proxyName, routerName, hre) {
-  const routerAddress = hre.deployer.deployment.general.contracts[routerName].deployedAddress;
+  const contracts = Object.values(hre.deployer.deployment.general.contracts);
+  const routerData = contracts.find((data) => data.isRouter);
+  const proxyData = contracts.find((data) => data.isProxy);
```

Comment:
Gets the data for the contracts using the new `isRouter` and `isProxy` markers on the deployment files. This avoids any errors and comptibilities with re-calculating the FQ names manually.